Enterprise-CMCS/macpro-onemac
# PI1.05 26321: Add Final Disposition to the Dashboard

Story: https://qmacbis.atlassian.net/browse/OY2-26321
Endpoint: See github-actions bot comment

### Details
CMS has asked that Package Withdrawn date get added to the dashboard for use with filters and reports.  We discovered that "Final Disposition" date is the Package Withdrawn date when the package is in Withdrawn status, so decided to add that column to the dashboard instead.

### Changes
- Because of weird date conversion issues, see "Implementation Notes," added the finalDispositionTimestamp to the package items and added enough milliseconds to be on the correct date in the eastern timezone.
- Added finalDispositionTimestamp to the dashboard as a hidden column.
- added filter details to make filters work as dictated

### Implementation Notes
The date being used is the SEA Tool STATUS_DATE on the STATE_PLAN record.  Now... that date is captured in the SEA Tool application as a mm-dd-yyyy date and sent to us as a timestamp... however, the timestamp is for GMT, so in any date conversion that uses local timezone, it is a day off.  We get around this in other places by converting the date on the back end (which uses GMT), but in order to use the date filter widget, we need a timestamp, so I've added enough milliseconds to the "timestamp" to have it be on the correct date in either GMT or Eastern.

### Test Plan
1. Login as a state user
2. Navigate to Dashboard
3. Verify Final Disposition is not a default column
4. Click "Show/Hide Columns"
5. Select Final Disposition
6. Verify the column is added to the dashboard and contains the correct value (can compare to details page)
7. Verify filters work as expected
8. Verify saving of filters works as expected
9. Verify exports are correct
10. Do the same tests with a CMS user
